src/components/draggable/draggable-item.component.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { useDrop, useDrag } from 'react-dnd';
+import PropTypes from 'prop-types';
+import { StyledDraggableItem } from './draggable-item.style';
+
+const DraggableItem = ({
+  id, findItem, moveItem, getOrder, children
+}) => {
+  const oIndex = findItem(id).index;
+  const [{ isDragging }, drag] = useDrag({
+    item: { type: 'draggableItem', id, oIndex },
+    collect: monitor => ({
+      isDragging: monitor.isDragging()
+    }),
+    end: (dropResult, monitor) => {
+      const { id: droppedId, originalIndex } = monitor.getItem();
+      const didDrop = monitor.didDrop();
+      if (!didDrop) {
+        moveItem(droppedId, originalIndex);
+      }
+
+      getOrder();
+    }
+  });
+
+  const [, drop] = useDrop({
+    accept: 'draggableItem',
+    canDrop: () => false,
+    hover({ id: draggedId }) {
+      if (draggedId !== id) {
+        const { index: overIndex } = findItem(id);
+        moveItem(draggedId, overIndex);
+      }
+    }
+  });
+
+  return (
+    <StyledDraggableItem
+      data-element='draggable'
+      isDragging={ isDragging }
+      ref={ node => drag(drop(node)) }
+    >
+      {children}
+    </StyledDraggableItem>
+  );
+};
+
+DraggableItem.propTypes = {
+  /**
+   * The id of the `DraggableItem`.
+   *
+   * Use this prop to make `Draggable` works
+    */
+  id: PropTypes.oneOfType([
+    PropTypes.number,
+    PropTypes.string
+  ]).isRequired,
+  /** The content of the component. */
+  children: PropTypes.node.isRequired,
+  getOrder: PropTypes.func,
+  findItem: PropTypes.func,
+  moveItem: PropTypes.func
```

Comment:
all these props should be required, if they're not defined, an error will be triggered

Reply:
You are right about `findItem` and `moveItem`. I have already change it. Also, I've turned off eslint validation `/* eslint-disable react/default-props-match-prop-types */` to avoid showing error in this specific case.
I would like to stay with `getOrder` as an optional prop because this prop is not required to make the component works.